source/common/api/BUILD

```diff
@@ -22,8 +22,16 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "os_sys_calls_lib",
-    srcs = ["os_sys_calls_impl.cc"],
-    hdrs = ["os_sys_calls_impl.h"],
+    srcs = ["os_sys_calls_impl.cc"] + select({
+        "@bazel_tools//src/conditions:linux_x86_64": ["os_sys_calls_impl_linux.cc"],
```

Comment:
FTR, this breaks the Google import since we don't have `@bazel_tools` conditional in this form internally. I'd prefer to stick with `//bazel:BUILD` conditionals, I will do an update PR for this.